lib/stanford-mods/date_parsing.rb

```diff
@@ -85,6 +86,55 @@ def self.facet_string_for_century(date_str)
         end
       end
 
+      # get String sortable value for B.C. if we have  B.C. pattern
+      #  note that these values must *lexically* sort to create a chronological sort.
+      #  We know our data does not contain B.C. dates older than 999, so we can make them
+      #  lexically sort by subtracting 1000.  So we get:
+      #    -700 for 300 B.C., -750 for 250 B.C., -800 for 200 B.C., -801 for 199 B.C.
+      # @param [String] date_str String containing B.C.
+      # @return [String, nil] String sortable -ddd if B.C. in pattern; nil otherwise
+      def self.sortable_year_for_bc(date_str)
+        bc_matches = date_str.match(/(\d{1,4}).*B\.C\./) if date_str
+        return ($1.to_i - 1000).to_s if bc_matches
+      end
+
+      # get single facet value for B.C. if we have  B.C. pattern
+      # @param [String] date_str String containing B.C.
+      # @return [String, nil] ddd B.C.  if ddd B.C. in pattern; nil otherwise
+      def self.facet_string_for_bc(date_str)
+        bc_matches = date_str.match(/\d{1,4}.*B\.C\./) if date_str
```

Comment:
I thought about this, but it didn't seem like it would make the code clearer to abstract it.  If it was more than one line, I would be more likely to do so.  What do you think?  Would it actually make it easier or harder to follow what is going on in the code?